mongo/integration/unified/server_api_options_test.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) MongoDB, Inc. 2017-present.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License. You may obtain
+// a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+package unified
+
+import (
+	"fmt"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+// ServerAPIOptions is a wrapper for *options.ServerAPIOptions. This type implements the bson.Unmarshaler interface
+// to convert BSON documents to a ServerAPIOptions instance.
+type ServerAPIOptions struct {
+	*options.ServerAPIOptions
+}
+
+type ServerAPIVersion = options.ServerAPIVersion
+
+var _ bson.Unmarshaler = (*ServerAPIOptions)(nil)
+
+func (to *ServerAPIOptions) UnmarshalBSON(data []byte) error {
+	var temp struct {
+		ServerAPIVersion  ServerAPIVersion       `bson:"version"`
```

Comment:
All of the fields besides `Extra` should be pointers. The code later on that translates these fields to a `ServerAPIOptions` instance should do nil checks and use the corresponding `Set*` functions on the options type:

```
if temp.DeprecationErrors != nil {
    to.SetDeprecationErrors(*temp.DeprecationErrors)
}
...
```

That way, we're only setting the options that were actually specified by the test rather than setting zero values when an option isn't specified.

Reply:
Discussed offline, made the two bools pointers, but left the `ServerAPIVersion` as is since it's just a string. Added the two nil checks and `Set...()` functions for the two bools.